### PR TITLE
2049 - New Validation added for parameters checking casing matches.

### PR DESCRIPTION
**Pull request by ralphdoe**
2049 - New Validation added for parameters checking casing matches, test added.

### Description
New Validation Added for this ticket: Path parameter validation should be smart about improper casing #2049 
When parameters have the same characters but different casing scenarios compared with path elements you will be able to see this message: 
In this case we have a Petid parameter, and petid as path parameter.
Parameter names are case-sensitive. The parameter named Petid does not match the case used in the 
path /pets/{petId}.

### Motivation and Context

This fixes the issue: https://github.com/swagger-api/swagger-editor/issues/2049

### How Has This Been Tested?
I added a test case in paths.js to validate this, and also did manual testing with swagger-editor tool.

### My PR contains... 
- [ ] No code changes (`src/` is unmodified: changes to documentation, CI, metadata, etc.)
- [ ] Dependency changes (any modification to dependencies in `package.json`)
- [x] Bug fixes (non-breaking change which fixes an issue)
- [x] Improvements (misc. changes to existing features)
- [ ] Features (non-breaking change which adds functionality)

### My changes...
- [ ] are breaking changes to a public API (config options, System API, major UI change, etc).
- [ ] are breaking changes to a private API (Redux, component props, utility functions, etc.).
- [ ] are breaking changes to a developer API (npm script behavior changes, new dev system dependencies, etc).
- [x] are not breaking changes.

### Documentation
- [x] My changes do not require a change to the project documentation.
- [ ] My changes require a change to the project documentation.
- [ ] If yes to above: I have updated the documentation accordingly.

### Automated tests
- [ ] My changes can not or do not need to be tested.
- [x] My changes can and should be tested by unit and/or integration tests.
- [x] If yes to above: I have added tests to cover my changes.
- [ ] If yes to above: I have taken care to cover edge cases in my tests.
- [x] All new and existing tests passed.

### CODE REVIEW

File: test/unit/plugins/validate-semantic/2and3/paths.js [hunk 1]
```diff
@@ -426,6 +426,36 @@ describe("validation plugin - semantic - 2and3 paths", () => {
...
+              }
+            }
+          }
+  
+          return validateHelper(spec)
+            .then(system => {
+              const allErrors = system.errSelectors.allErrors().toJS()
+              const firstError = allErrors[0]
+              expect(allErrors.length).toEqual(1)
+              expect(firstError.message).toEqual("Parameter names are case-sensitive. The parameter named \"myParam\" does not match the case used in the path \"/{myParam}\".")
```
[the remark this answers is not in the document]
Comment (ralphdoe) [reply test/unit/plugins/validate-semantic/2and3/paths.js]: Resolved.
